sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/resource/SimpleHttpClient.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.aws.resource;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+/** A simple HTTP client based on OkHttp. Not meant for high throughput. */
+final class SimpleHttpClient {
```

Comment:
Total NIT: as it's bound with OkHttp for good and bad I'd name it: "SimpleOkHttpClient" to note that.

Reply:
Might end up swapping OkHttp back to HTTPUrlConnection or something else later, next time won't have to change class name references everywhere by using a more generic name here :)